lib/web/files.go

```diff
@@ -68,6 +76,22 @@ func (h *Handler) transferFile(w http.ResponseWriter, r *http.Request, p httprou
 		proxyHostPort: h.ProxyHostPort(),
 	}
 
+	mfaReq, err := clt.IsMFARequired(r.Context(), &proto.IsMFARequiredRequest{
+		Target: &proto.IsMFARequiredRequest_Node{
+			Node: &proto.NodeLogin{
+				Node:  p.ByName("server"),
+				Login: p.ByName("login"),
+			},
+		},
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if mfaReq.Required && query.Get("webauthn") == "" {
+		return nil, trace.BadParameter("MFA required for file transfer.")
```

Comment:
If MFA is required, and MFA wasn't provided, should this be `trace.AccessDenied` instead?

Reply:
Yeah that makes sense. My original idea for `BadParameter` was "you didnt send in the parameter that was needed". I think I was in dev mode of "I need to send this parameter" but from the actual user's perspective, it makes more sense to say "You don't have access" . Thanks 👍